src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs

```diff
@@ -193,30 +184,32 @@ internal static void ReadMetadata(SqlConnection sqlConn, List<ObjectMetadata> me
                         var objectType = reader[2] as string;
 
                         MetadataType metadataType;
-                        string metadataTypeName;
                         if (objectType.StartsWith("V"))
                         {
                             metadataType = MetadataType.View;
-                            metadataTypeName = "View";
                         }
                         else if (objectType.StartsWith("P"))
                         {
                             metadataType = MetadataType.SProc;
-                            metadataTypeName = "StoredProcedure";
                         }
                         else
                         {
                             metadataType = MetadataType.Table;
-                            metadataTypeName = "Table";
                         }
-
-                        metadata.Add(new ObjectMetadata
+                        // remove temporary tables and stored procedures
+                        // since every temp object has a '#' in front of it
+                        if (objectName[0] == '#') 
```

Comment:
>if (objectName[0] == '#')  [](start = 24, length = 26)

not sure if I'm missing something.  Where do we add metadata for objects that don't start with '#'?

Reply:
We don't, that's the change. The OE doesn't show temporary objects (objects starting with '#'), and thus this change to make the dashboard consistent with OE. Otherwise, dashboard shows temporary objects if they exist, while OE doesn't.

Reply:
@abist I'm still not following.  Currently we do add metadata objects that **don't** start with a '#' to the dashboard.  With this change I can't see where we're adding these objects.

Reply:
This is a PR to **NOT** to add those objects. Those objects start with "#" are temporary. As of now, they are being shown in the dashboard. If you see the bug, we want similar behavior in OE and the dashboard. The OE **doesn't** show these objects with "#".

If it's still not clear, we can sync offline tomorrow.